### PR TITLE
Add private data chaincode-typescript

### PR DESCRIPTION
**Pull request by YunusEmreKarakose**
Hi,

I added a typescript implementation of private data chaincode. I tested it with the private data tutorial (https://hyperledger-fabric.readthedocs.io/en/latest/private_data_tutorial.html) and everything works fine.

I can make changes, etc., if needed.

Kind regards

### CODE REVIEW

**Comment by ryjones** [comment]
Thank you, @YunusEmreKarakose . [Please see instructions here on how to fix the DCO issue](https://github.com/hyperledger/fabric-samples/pull/1022/checks)

**Comment by denyeart** [comment]
Thanks for the contribution. This pull request includes already merged commits so it is difficult to see exactly what is new. Please rebase against fabric-samples main branch and squash your commits so that there is a single new commit to review.

**Comment by ryjones** [comment]
@denyeart [I'm not sure why this was cancelled](https://github.com/hyperledger/fabric-samples/actions/runs/4809936468/jobs/8564968982?pr=1022)

**Comment by denyeart** [comment]
@Mergifyio rebase

**Comment by YunusEmreKarakose** [comment]
@ryjones Is there a problem? If there is, I can fix it. Sorry, I messed up with git usage.

**Comment by ryjones** [comment]
@YunusEmreKarakose I did the rebase and squash. @denyeart please review

File: asset-transfer-private-data/chaincode-typescript/package.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
+    "name": "asset-transfer-basic",
+    "version": "1.0.0",
+    "description": "Asset Transfer Basic contract implemented in TypeScript",
```
Comment (denyeart) [inline asset-transfer-private-data/chaincode-typescript/package.json:4]: Please update name and description

File: asset-transfer-private-data/chaincode-typescript/package.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    },
+    "scripts": {
+        "lint": "tslint -c tslint.json 'src/**/*.ts'",
+        "pretest": "npm run lint",
+        "test": "nyc mocha -r ts-node/register src/**/*.spec.ts",
+        "start": "set -x && fabric-chaincode-node start",
+        "build": "tsc",
+        "build:watch": "tsc -w",
+        "prepublishOnly": "npm run build",
+        "docker": "docker build -f ./Dockerfile -t asset-transfer-basic .",
```
Comment (denyeart) [inline asset-transfer-private-data/chaincode-typescript/package.json:19]: Update

**Review by denyeart — APPROVED** [review]
Thank you for the contribution!

**Pull request merged**